src/features/wallet/root/ActivityCardHeader.tsx

```diff
@@ -1,32 +1,63 @@
 import React from 'react'
+import { useTranslation } from 'react-i18next'
+import { useActionSheet } from '@expo/react-native-action-sheet'
 import CardHandle from './CardHandle'
 import Box from '../../../components/Box'
 import Text from '../../../components/Text'
 import CarotDown from '../../../assets/images/carot-down.svg'
-import { triggerNavHaptic } from '../../../utils/haptic'
+import { FilterKeys, FilterType } from './walletTypes'
 import TouchableOpacityBox from '../../../components/TouchableOpacityBox'
+import { Font } from '../../../theme/theme'
 
-const ActivityCardHeader = () => {
-  const handlePress = () => {
-    triggerNavHaptic()
+type Props = {
+  filter: FilterType
+  onFilterChanged: (filter: FilterType) => void
+}
+
+const ActivityCardHeader = ({ filter, onFilterChanged }: Props) => {
```

Comment:
Not sure if it would be a lot of work, but we will need this in other places too so making it generic would be awesome.

Maybe instead of using specific filters we could make it a generic string array? We could also change it if we want when we need to use it in another place?

Reply:
yeah, I'm with you on that. For now, I'd rather leave it as a todo

Reply:
Yeah sounds good